### PR TITLE
[Core] Kratos-Components minor cleaning

### PR DESCRIPTION
**Pull request by philbucher**
this PR cleans up the KratosComponents to separate the cleaning from the new implementation in #3432

### CODE REVIEW

File: kratos/includes/kratos_components.h [hunk 1]
```diff
@@ -21,7 +20,6 @@
 #include <map>
 
 // External includes
-#include <boost/ref.hpp>
```
Comment (philbucher) [inline kratos/includes/kratos_components.h:24]: this is no longer needed with c++11 => https://www.boost.org/doc/libs/1_66_0/libs/numeric/odeint/doc/html/boost_numeric_odeint/odeint_in_detail/using_boost__ref.html

anyway it was only used in commented code

File: kratos/includes/kratos_components.h [hunk 2]
```diff
@@ -76,31 +73,21 @@ class KRATOS_API(KRATOS_CORE) KratosComponents
...
-//     }
-
-
     static TComponentType const& Get(std::string const& Name)
     {
-        typename ComponentsContainerType::iterator i =  msComponents.find(Name);
-        if(i == msComponents.end())
+        auto it_comp =  msComponents.find(Name);
+        if(it_comp == msComponents.end())
           KRATOS_THROW_ERROR(std::invalid_argument, "The component is not registered!", Name);
```
Comment (philbucher) [inline kratos/includes/kratos_components.h:89]: this is not updated bcs it is changes in #3432

**Review by pooyan-dadvand — DISMISSED** [review]
Thanks! 👍

**Comment by philbucher** [comment]
@jcotela I reverted making the print fcts static

**Comment by philbucher** [comment]
@pooyan-dadvand could you please reapprove?

**Review by pooyan-dadvand — APPROVED** [review]
:+1:

**Pull request merged**